### PR TITLE
Validating Dtypes from String

### PR DESCRIPTION
**Pull request by fschrader1992**
This PR
- adds validations fro property values with dtype string. It checks, whether possible other dtypes might have been detected as string instead and issues a warnings.
- adds corresponding tests.
- adds new xml, yaml and json files to test respective validations.

Checks for Code Quality:
- [x] Pylint
- [x] Rebase from GNODE Master
- [x] Repeat Local Tests Python3.8
- [x] Repeat Local Tests Python2.7

### CODE REVIEW

File: test/test_validation.py [hunk 1]
```diff
@@ -196,3 +196,31 @@ def test_prop_string_values(self):
                               values=['(39.12; 89; 67.19)', '(39.12; 78; 67.19)', '(39.12)'])
         self.assertError(validate(prop9),
                          'Dtype of property "Coos" currently is "string", but might fit dtype "3-tuple"!')
+
+    def test_load_xml(self):
```
Comment (mpsonntag) [inline test/test_validation.py]: Please split this test into individual tests for a) undefined section type and b) properties with undefined dtypes. Same for the `json` and `yaml` tests.

File: test/resources/integration.xml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  </section>
+  <section>
+    <id>27ca5a42-fc00-4342-af67-45d014c92d90</id>
+    <type>test</type>
+    <name>sec1</name>
+    <property>
+      <id>b237006e-a584-469d-adbc-e30dae9c4678</id>
+      <name>members</name>
+      <value>[-13,101,-11,0,-8]</value>
+      <type>string</type>
```
Comment (mpsonntag) [inline test/resources/integration.xml]: Please refactor the test file; same for `json` and `yaml`:
a) please use a different naming scheme to make it clear to which test these files belong e.g. `validation_dtypes` and `validation_format`.
b) please move the content for missing section type to their own files and create distinct tests to make them independent.
c) please add as discussed two main sections with subsections. One with `type=string` everywhere and one with `type` missing completely. Add additional assertions accordingly to the same tests. It seems redundant, but I really would like to have both individual cases covered.

File: test/test_validation.py [hunk 3]
```diff
@@ -206,7 +207,9 @@ def test_load_xml(self):
         - Properties with undefined dtypes
         """
 
-        doc = odml.load("./resources/integration.xml")
+        dir_path = os.path.dirname(os.path.realpath(__file__))
```
Comment (mpsonntag) [inline test/test_validation.py]: Just a minor comment: `dir_path = ...` could be moved to the `setUp()` method and used from there.

**Comment by fschrader1992** [comment]
Checks for Code Quality:
- [x] Pylint
- [x] Rebase from GNODE Master
- [x] Repeat Local Tests Python3.8
- [x] Repeat Local Tests Python2.7

**Comment by fschrader1992** [comment]
Checks for Code Quality:
- [x] Pylint
- [x] Rebase from GNODE Master
- [x] Repeat Local Tests Python3.8
- [x] Repeat Local Tests Python2.7

**Pull request merged**